File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.java
class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> a[0] == b[0] ? a[1] - b[1] : a[0] - b[0]);
        for(int i=0; i<mat.length; i++){
            int cnt = 0;
            for(int j=0; j<mat[0].length; j++){
                cnt += mat[i][j];
            }
            pq.offer(new int[]{cnt, i});
        }
        int[] ans = new int[k];
        for(int i=0; i<k; i++){
            int[] x = pq.poll();
            ans[i] = x[1];
        }
        return ans;
    }
}

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
        for(int i=0; i<mat.size(); i++){
            int cnt = 0;
            for(int j=0; j<mat[0].size(); j++){
                cnt += mat[i][j];
            }
            pq.push({cnt, i});
        }
        vector<int> ans;
        for(int i=0; i<k; i++){
            auto x = pq.top(); pq.pop();
            ans.push_back(x.second);
        }
        return ans;
    }
};